### PR TITLE
Fix a doc. cloneWithProps clones a ReactElement.

### PR DESCRIPTION
**Pull request by koba04**
React.addons.cloneWithProps clones a ReactElement.
This pull request makes it more expressly.

### CODE REVIEW

**Comment by jsfb** [comment]
This looks good.  Before we merge, can you change it to use a lowercase `element` (instead of `ReactElement`) anywhere we were previously using a lowercase `component`.  The reason is that it reads better and makes it seem less foreign/intimidating to people who are not yet familiar with all the formal terminology.

File: src/utils/cloneWithProps.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ var CHILDREN_PROP = keyOf({children: null});
  * Sometimes you want to change the props of a child passed to you. Usually
  * this is to add a CSS class.
  *
- * @param {object} child child component you'd like to clone
+ * @param {object} child child ReactElement you'd like to clone
```
Comment (vsiao) [inline src/utils/cloneWithProps.js]: I suppose this should be `@param {ReactElement} element`, and the same for the return type below

Comment (jsfb) [inline src/utils/cloneWithProps.js]: Yeah, sounds right to me.  cc @zpao

Comment (jsfb) [inline src/utils/cloneWithProps.js]: I'll fix in src https://github.com/facebook/react/pull/3208

**Comment by koba04** [comment]
@jsfb OK, I've fixed it!

File: src/utils/cloneWithProps.js (hunk 1 above)
Comment (koba04) [inline src/utils/cloneWithProps.js]: Sounds good.

**Comment by jsfb** [comment]
@koba04 Looks great, thanks!

**Comment by koba04** [comment]
Thanks!!

**Pull request merged**